src/jdk.jshell/share/classes/jdk/jshell/execution/RemoteExecutionControl.java

```diff
@@ -63,13 +65,19 @@ public static void main(String[] args) throws Exception {
         InputStream inStream = socket.getInputStream();
         OutputStream outStream = socket.getOutputStream();
         Map<String, Consumer<OutputStream>> outputs = new HashMap<>();
-        outputs.put("out", st -> System.setOut(new PrintStream(st, true)));
-        outputs.put("err", st -> System.setErr(new PrintStream(st, true)));
+        outputs.put("out", st -> System.setOut(new PrintStream(st, true, getPrintStreamCharset(System.out))));
+        outputs.put("err", st -> System.setErr(new PrintStream(st, true, getPrintStreamCharset(System.err))));
         Map<String, Consumer<InputStream>> input = new HashMap<>();
         input.put("in", System::setIn);
         forwardExecutionControlAndIO(new RemoteExecutionControl(), inStream, outStream, outputs, input);
     }
 
+    // Get PrintStream charset
+    private static Charset getPrintStreamCharset(PrintStream ps) {
+        return Charset.forName(new OutputStreamWriter(ps).getEncoding(),
+            Charset.defaultCharset());
+    }
+
```

Comment:
I'd prefer not to introduce this static method, but just embed it into each location. The other comment in `AbstractTerminal.java` applies here too.